Спринт 9

AnnaCher22 opened the pull request:
Задачи на спринт
Тема 1. Стили и темы
Тема 2. Context и Intent
Тема 3. EditText и TextWatcher
Тема 4. Хранение данных. Начало

AnnaCher22 committed:
Спринт 9
Задачи на спринт
Тема 1. Стили и темы
Тема 2. Context и Intent
Тема 3. EditText и TextWatcher
Тема 4. Хранение данных. Начало

ArturNurtdinov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,76 +1,38 @@
         app:cornerRadius="@dimen/button_radius"
-        app:icon="@drawable/find"
+        app:icon="@drawable/set"
         app:iconGravity="textStart"
```
Свойства `iconGravity`, `cornerRadius`, `iconTint` можно вынести в стиль тоже, так как они относятся к отображению вью, а не к её расположению на экране

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,88 @@
+        clearSearchButton.setOnClickListener {
+            searchEditText.text.clear()
+            searchEditText.requestFocus()
+            showKeyboard(searchEditText)
```
Здесь видимо опечатка, т.к. нужно скрывать клавиатуру:
```suggestion
            searchEditText.clearFocus()
            hideKeyboard()
```

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,88 @@
+    // Сохранение текста перед уничтожением активити
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString("SEARCH_QUERY", searchQuery)
```
Ключи для передачи данных лучше вынести в константу, т.к. они используются в более чем 1 месте. Ещё один плюс вынесения в константы - в будущем легче будет этот экран переиспользовать

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -12,10 +13,39 @@ class SettingsActivity : AppCompatActivity() {
+                data = Uri.parse("mailto:${getString(R.string.support_email)}")
+                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.email_subject))
+                putExtra(Intent.EXTRA_TEXT, getString(R.string.email_body))
+            }
```
Для корректного заполнения полей необходимо настроить интент следующим образом:
```suggestion
        val emailIntent = Intent(Intent.ACTION_SENDTO).apply {
                data = "mailto:".toUri()
                putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.support_email)))
                putExtra(Intent.EXTRA_SUBJECT, getString(R.string.email_subject))
                putExtra(Intent.EXTRA_TEXT, getString(R.string.email_body))
            }
```

ArturNurtdinov commented on app/src/main/res/values/colors.xml:
```diff
@@ -4,4 +4,18 @@
+    <color name="icon_color_day">#AEAFB4</color> <!-- Цвет для дня -->
+    <color name="icon_color_night">#FFFFFF</color> <!-- Цвет для ночи -->
+
+    <color name="icon_color">#AEAFB4</color> <!-- Серый для дня -->
```
Комментарии здесь излишни, так как в студии можно смотреть превью цвета

ArturNurtdinov reviewed (commented):
1. Отступы в заголовке на главном экране не соответствуют макетам:
![image](https://github.com/user-attachments/assets/5f97bbf2-0ed0-40b7-af89-17ed0d9dc95d)
Макет:
![image](https://github.com/user-attachments/assets/d0996c0c-e921-45bb-8dd4-15025fb874ab)
2. Поле ввода и кнопка "Назад" не соответствуют макетам на экране поиска:
![image](https://github.com/user-attachments/assets/e21b02da-2568-46fe-8fce-124f11d38c21)
Макет:
![image](https://github.com/user-attachments/assets/c03138c0-4c81-4053-a825-b75ec87b2cfe)
3. Кнопка "Назад" и заголовок на экране настроек не соответствуют макетам:
![image](https://github.com/user-attachments/assets/64b4c8ea-b64d-4f26-8b9c-b59ed828d93c)
Макет:
![image](https://github.com/user-attachments/assets/a07bed63-4e1b-4f75-8f13-c7915b5d515f)

AnnaCher22 committed:
Спринт 9
Задачи на спринт
Тема 1. Стили и темы
Тема 2. Context и Intent
Тема 3. EditText и TextWatcher
Тема 4. Хранение данных. Начало

AnnaCher22 committed:
Спринт 9
Задачи на спринт
Тема 1. Стили и темы
Тема 2. Context и Intent
Тема 3. EditText и TextWatcher
Тема 4. Хранение данных. Начало

ArturNurtdinov reviewed (commented):
Остались два критических замечания:
1. Поле ввода не соответствует макетам в обеих темах приложения:
![image](https://github.com/user-attachments/assets/d5cf325a-78d2-44f8-885c-1a2e75e61522)
![image](https://github.com/user-attachments/assets/008b187e-7a69-49d0-b3f2-8a28b7c70662)
Макеты:
![image](https://github.com/user-attachments/assets/fa4c4de2-6f39-49b8-b481-19f7907609e4)
![image](https://github.com/user-attachments/assets/41cb94d9-79ca-466f-92b2-719b121af785)
2. Кнопка "Назад" на экране настроек не отцентрирована:
![image](https://github.com/user-attachments/assets/3b2c24e3-03d2-4954-a369-d4c4e885b0be)
Макет:
![image](https://github.com/user-attachments/assets/a3905da0-269e-4a0a-89d6-58d180e5f852)
